api/domain/index.js

```diff
@@ -0,0 +1,102 @@
+import { Router } from "express";
+import {
+  validateCustomFieldCreation,
+  validateCustomFieldUpdate,
+  validateDomainSettingsUpdate,
+} from "../../validation/domain";
+import { isValidModelId } from "../../validation/validationUtils";
+import Domain from "../../models/domain";
+
+const router = new Router();
+
+const validateDomain = (req, res, next) => {
+  if (isValidModelId(req.params.domainId)) {
+    Domain.findById(req.params.domainId)
+      .then(domain => {
+        if (domain !== null) {
+          if (domain._id.equals(req.user.domain)) {
+            next();
+          } else {
+            return res.status(403).json({
+              errors: {
+                message: "You don't have necessary rights to access this domain",
+              },
+            });
+          }
+        } else {
+          return res.status(404).json({
+            errors: {
+              message: "Domain with provided id is not found",
+            },
+          });
+        }
+      });
+  } else {
+    return res.status(404).json({ errors: { message: "Provided domain id is not valid" } });
+  }
+};
+
+const domainMembersMiddlewares = [validateDomain];
```

Comment:
redundant array, just use middleware function in routes instead

Reply:
we've added this for the case if we have more middlewares in future